scripts/keymaker-create-account-for-iam-user

```diff
@@ -1,22 +1,55 @@
-#!/bin/bash -e
+#!/bin/bash
+
+# TODO: what to do if /etc/ssh/sshd_config isn't the running SSHD config file
+# TODO: more graceful handling of login banner (this is a stub) - libssh?
+# TODO: test user/group creation on: Amazon Linux, Suse, CentOS
 
 EX_TEMPFAIL=75
 EX_NOPERM=77
 
-if getent passwd "$PAM_USER" >/dev/null 2>&1; then
-    # Terminate the PAM authentication stack. The SSH client will fail since the user didn't supply a valid public key.
-    exit $EX_NOPERM
+PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+SSHD_CONFIG="/etc/ssh/sshd_config"
+BANNER_FILE=`grep -e "^Banner." $SSHD_CONFIG | awk '{print $2}'`
+
+# Get the login banner
+if [ -e $BANNER_FILE ] ; then
+    BANNER=`cat $BANNER_FILE`
 else
-    # Create the user, then terminate the PAM authentication stack. The SSH client will fail, and the user will need to try again.
-    # TODO: figure out how to display info banner
-    # Verify that the IAM user exists.
-    keymaker get_authorized_keys "$PAM_USER" >/dev/null
+    BANNER=''
+fi
+
+create_account () {
     NEW_UID=$(keymaker get_uid "$PAM_USER")
     adduser "$PAM_USER" --disabled-password --gecos "$PAM_USER" --uid "$NEW_UID"
-    for group in $(keymaker get_groups "$PAM_USER"); do
-        usermod --append --groups "$group" "$PAM_USER" || echo "$0: Error while adding user to group"
+    groups=`keymaker get_groups "$PAM_USER"`
+    for group in $groups ; do
+        getent group | egrep "^$group:" >/dev/null
+        if [ "$?" == "1" ] ; then
+            echo "Creating group: $group"
+            groupadd $group
```

Comment:
There are a couple issues when logging in with a new account:

<pre>(local) TOPD-061012:keymaker rmcdonough$ ssh rich@jump.dcerberus.com
INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 169.254.169.254
INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 169.254.169.254
INFO:botocore.credentials:Found credentials from IAM Role: jump
INFO:boto3.resources.action:Calling iam:get_user with {u'UserName': 'rich'}
INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): iam.amazonaws.com
Adding user `rich' ...
Adding new group `rich' (19664) ...
Adding new user `rich' (19664) with group `rich' ...
Creating home directory `/home/rich' ...
Copying files from `/etc/skel' ...
INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 169.254.169.254
INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 169.254.169.254
INFO:botocore.credentials:Found credentials from IAM Role: jump
INFO:boto3.resources.collection:Calling paginated iam:listPassword: 
...CTRL+C...</pre>

I can login just fine on the second attempt though, after the account is created though. This happens with the master version of ```keymaker-create-account-for-iam-user ``` as well.